content/posts/0.28-release.md

```diff
@@ -0,0 +1,254 @@
+
+## New matplotlib backend for Pyodide
+
+For years, Pyodide relied on a custom Matplotlib backend (`wasm_backend`) to
+render plots directly in your browser. This backend was developed by the creator
+of Pyodide, Michael Droettboom, who was also a core developer of matplotlib.
+
+However, it hadn't been maintained for a long time, and without dedicated
+expertise among our core developers, it became increasingly incompatible with
+newer Matplotlib versions. This made it tough to keep up with new features and
+critical bug fixes.
+
+In this release, we have deprecated these custom backend, and replaced it with
+the new patched version of the `webagg` backend, which is one of the official
+matplotlib backends. This new backend provides a more stable and feature-rich
+experience for rendering matplotlib plots in the browser.
+
+A huge thank you to [Ian Thomas](https://github.com/ianthomas23), a core
+Matplotlib developer and JupyterLite maintainer, who is now maintaining this new
+backend.
+
+## Other improvements
+
+### Standardized Package Loading with Runtime Paths
+
+This release introduces support for runtime paths in Emscripten modules, which
+allows us to correctly locate shared library dependencies.
+
+If a Python binary extension `somebinmod.so` depends on a shared library,
+`libsomedep.so`, this information will be included in the dynamic loader section
+of `somebindmod.so`. The dynamic loader will search for `libsomedep.so` on the
+`LD_LIBRARY_PATH`.
+
+Python wheels vendor their shared libraries, so if `somebinmod.so` is contained
+in `somepackage` then `somepackage.whl` will include a folder called
+`somepackage.libs` with `libsomedep.so` inside. However, `somepackage.libs`
+cannot be added to the `LD_LIBRARY_PATH` because we only want to search that
+directory when opening shared libraries in `somepackage`.
+
+Instead, each dynamic library has its own dependency search path called the
+runtime path. This is information included in the dynamic loader section of a
+shared library indicating where its dependencies should be located. In this
+case, `somebinmod.so` would have an entry saying to look in `somepackage.libs`
+for its dependencies.
+
+Previously, WebAssembly shared libraries files did not support runtime paths, so
+we had to use a custom patch for the Emscripten dynamic loader to apply the rule
+that `somepackage.libs` should be searched when loading libraries from
+`somepackage`. This patch exposed us to extra bugs and prevented us from being
+able to upstream fixes to the Emscripten dynamic loader. It also forced us to
+load dynamic libraries eagerly rather than lazily.
+
+We added runtime path to the WebAssembly specification for the shared library
+format, to the llvm WebAssembly object parser and linker, and to the `emcc`
+linker. We also updated the dynamic loader to use this information. Finally, we
+updated `pyodide-build` to emit the new runtime path data and we removed our
+patch to the dynamic loader.
```

Comment:
Does it make sense to leave a mention or footnote about `auditwheel-emscripten` here? I suppose some people could have questions about how we repair our wheels.

Reply:
I think it's too much detail for a release note.